code/modules/admin/admin_fax_panel.dm

```diff
@@ -0,0 +1,134 @@
+
+	/// Default name of fax. Used when field with fax name not edited.
+	var/sending_fax_name = "Secret"
+	/// Default name of paper. paper - bluh-bluh. Used when field with paper name not edited.
+	var/default_paper_name = "Standart Report"
+
+/datum/fax_panel_interface/New()
+	//Get all faxes, and save them to our list.
+	for(var/obj/machinery/fax/fax in GLOB.machines)
+		available_faxes += fax
+	//Get all stamps
+	for(var/stamp in subtypesof(/obj/item/stamp))
+		var/obj/item/stamp/real_stamp = stamp 
+		if(length(initial(real_stamp.actions)) == 0)// try to remove chameleon, dont work.
+			stamp_list += list(list(initial(real_stamp.name), initial(real_stamp.icon_state)))
+	//Give our paper special status, to read everywhere.
+	fax_paper.request_state = TRUE
+
+//Maybe.. useless proc, but find fax like object what we need.
+/datum/fax_panel_interface/proc/get_fax_by_name(name)
+	if(!length(available_faxes))
+		return
+
+	for(var/obj/machinery/fax/potential_fax in available_faxes)
+		if(potential_fax.fax_name == name)
+			return potential_fax
+	return
+
+/datum/fax_panel_interface/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "AdminFax")
+		ui.open()
+
+/datum/fax_panel_interface/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/fax_panel_interface/ui_static_data(mob/user)
+	var/list/data = list()
+	data["faxes"] = list()
+	data["stamps"] = list()
+	for(var/stamp in stamp_list)
+		data["stamps"] += list(stamp[1]) // send only names.
+	for(var/obj/machinery/fax/another_fax in available_faxes)
+		data["faxes"] += list(another_fax.fax_name)
+	return data
+
+/datum/fax_panel_interface/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	if(..())
+		return
+
+	if(!check_rights(R_ADMIN))
+		return
+	
+	var/obj/machinery/fax/action_fax // fax what we find later
+
+	if(params["faxName"])
+		action_fax = get_fax_by_name(params["faxName"]) // that is
+	
+	switch(action)
+
+		if("jump") // jump on fax. maybe i shoul make follow action(?)
+			var/turf/fax_turf = get_turf(action_fax)
+			if(!fax_turf || !usr.client)
+				return
+			
+			usr.client.jumptoturf(fax_turf)
+		
+		if("preview") // see saved variant
+			if(!fax_paper)
+				return
+			
+			fax_paper.ui_interact(usr)
+		
+		if("save") // save paper
+			if(params["paperName"])
+				default_paper_name = params["paperName"]
+			if(params["fromWho"])
+				sending_fax_name = params["fromWho"]
+			
+			fax_paper.ui_status(usr, UI_CLOSE) // i wannd reload, or close and open UI, but i dont know how. need help
+
+			fax_paper.clear_paper()
+			var/stamp 
+
+			for(var/needed_stamp in stamp_list)
+				if(needed_stamp[1] == params["stamp"])
+					stamp = needed_stamp[2]
+					break
+			
+			fax_paper.name = "paper — [default_paper_name]"
+			fax_paper.add_raw_text(params["rawText"])
+
+			if(stamp)
+				fax_paper.add_stamp("paper121x54 [stamp]", params["stampX"], params["stampY"], params["stampR"], stamp)
```

Comment:
a part with div class name. 
![image](https://user-images.githubusercontent.com/78199449/200419709-e8b584d7-dfef-40f3-8074-64685bd6592f.png)

Reply:
Where do 121 and 54 come from? I want to make sure that whatever these relate to we can change without silently causing an error here.

Reply:
> Where do 121 and 54 come from? I want to make sure that whatever these relate to we can change without silently causing an error here.

i get it from (screenshot)
![image](https://user-images.githubusercontent.com/78199449/200425503-b7c1d893-7ae5-4a56-a5ba-de0b499fbb5f.png)
class name be paper121x54 + icon_state.

Reply:
But where is the number 121 and 54 there?

Reply:
> But where is the number 121 and 54 there?

name = "paper" size_id = 121x54, when i use this proc, i get this data

Reply:
But WHERE are those numbers coming from? Is there a way you can avoid hard coding them?

Reply:
> But WHERE are those numbers coming from? Is there a way you can avoid hard coding them?
![image](https://user-images.githubusercontent.com/78199449/200472303-a9323a43-69a3-485e-afca-df5770e20735.png)

Reply:
Ok, then we need to find some way to make this not a hardcoded string. Otherwise if anyone changes these stamp sizes your code will silently break.

Can you potentially grab the spritesheet directly and run `icon_class_name`?

Reply:
i can try

Reply:
> Ok, then we need to find some way to make this not a hardcoded string. Otherwise if anyone changes these stamp sizes your code will silently break.
> 
> Can you potentially grab the spritesheet directly and run `icon_class_name`?

take a look